### PR TITLE
#160371193 Edit Travel Request Notifications

### PR DESCRIPTION
**Pull request by albert85**
#### What does this PR do?
It sends a notification when a travel request is been edited.

#### Description of Task to be completed?
- Populate Notification pane with notifications.
- Open travel request details when notifications are clicked.

#### How should this be manually tested?
- Clone the repo - git clone https://github.com/andela/travel_tool_front.git
- Change into the project directory
- Run the server `make start`

#### Any background context you want to provide?
- Before testing, clone this repo [this](https://github.com/andela/travel_tool_back/pull/40) and follow the description to create `POST` a new travel request and edit `PUT`.
- Make sure your name is used as Manager on the backend and its is available on the database before you create a new request on the backend
- Start the server and navigate to the front end and click on the notification pane
- Click on the notifications to view the details

#### What are the relevant pivotal tracker stories?
[#160371193](https://www.pivotaltracker.com/story/show/160371193)

#### Screenshots (if appropriate)
screen shot 2018-09-19 at 9 43 59 pm

#### Questions:
N/A

### CODE REVIEW

**Comment by samkubai** [comment]
@albert85 please name this the correct way.

File: src/components/modal/Modal.jsx [hunk 1]
```diff
@@ -44,12 +53,19 @@ class Modal extends PureComponent {
       closeModal,
       title,
       modalId,
-      modalContentId
+      modalContentId,
+      params
     } = this.props;
+    
+    let url = location.pathname;
```
Comment (d3mola) [inline src/components/modal/Modal.jsx]: @albert85 this block of code looks similar to what we have from lines 24 - 28. Is there a reason why we're not abstracting that into a method?

File: src/components/notification-pane/NotificationContainer.jsx [hunk 2]
```diff
@@ -12,9 +12,11 @@ export default class NotificationContainer extends PureComponent {
         if(notification.notificationType === 'pending'){
           isPending = true;
         }
+        const{handleClick} = this.props;
```
Comment (d3mola) [inline src/components/notification-pane/NotificationContainer.jsx]: @albert85 you should pad destructed variables with spaces according to the engineering playbook. 
`const{handleClick} = this.props;` could be `const { handleClick } = this.props;`

File: src/components/notification-pane/NotificationItem.jsx [hunk 3]
```diff
@@ -1,21 +1,29 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
-
+import { Link } from 'react-router-dom';
 import './_notificationItem.scss';
 import readMessageIcon from '../../images/read-message.svg';
 import unreadMessageIcon from '../../images/unread-message.svg';
+// import socket from '../../helper/socket/socket';
```
Comment (d3mola) [inline src/components/notification-pane/NotificationItem.jsx]: @albert85 looks like we don't need this commented line

File: src/components/notification-pane/NotificationItem.jsx [hunk 4]
```diff
@@ -1,21 +1,29 @@
...
 
 export default class NotificationItem extends PureComponent {
   renderNotificationItemMetaInfo = () => {
-    const { isPending, notificationStatus } = this.props;
+    // const { sender, recepient, requestId, status, type } = this.state;
```
Comment (d3mola) [inline src/components/notification-pane/NotificationItem.jsx]: commented code here too

File: src/components/notification-pane/NotificationPane.jsx [hunk 5]
```diff
@@ -1,86 +1,80 @@
...
   componentDidMount() {
-    const { user, fetchNotifications } = this.props;
-    handleNotification(user && user.UserInfo.id);
-    fetchNotifications();
+    const { user, fetchUsersNotification } = this.props;
+    fetchUsersNotification(); 
+    handleManagerNotification(user && user.UserInfo.id);
   }
 
-  getNotifications() {
```
Comment (d3mola) [inline src/components/notification-pane/NotificationPane.jsx]: @albert85 why did you remove this method, I remember I created and used it.

File: src/components/notification-pane/NotificationPane.jsx [hunk 6]
```diff
@@ -1,86 +1,80 @@
...
+    const { onCloseNotificationPane, notifications } = this.props;
     const generalNotifications = [];
-    const pendingNotifications = [];
-    notifications.length && notifications.map(notification => {
-      if (notification.notificationType === 'general') {
+    let pendingNotifications = notifications.length && notifications.filter(notification => {
+      if(notification.notificationType !== 'pending'){
         generalNotifications.push(notification);
-      } else if(notification.notificationType === 'pending') {
-        pendingNotifications.push(notification);
```
Comment (d3mola) [inline src/components/notification-pane/NotificationPane.jsx]: @albert I refactored this block of code to this because I had some issues with the value of pendingNotififcations, I kept getting the number `zero` instead of an empty array when it's empty. I see that you have refactored to what we had earlier. Could you provide clarity on why you did that|?

File: src/components/notification-pane/NotificationPane.jsx [hunk 7]
```diff
@@ -1,86 +1,80 @@
...
-  notifications: PropTypes.array,
-  fetchNotifications: PropTypes.func.isRequired,
-  user: PropTypes.object,
+  fetchUsersNotification: PropTypes.func.isRequired,
+  user: PropTypes.object.isRequired,
+  notifications: PropTypes.arrayOf(PropTypes.object)
 };
 
 NotificationPane.defaultProps = {
-  notifications: [],
```
Comment (d3mola) [inline src/components/notification-pane/NotificationPane.jsx]: @albert85 , didn't refactoring this part cause you to have warnings in the console? I ask because I added that and I see that you refactored.

File: src/redux/actionCreator/notificationsActions.js [hunk 8]
```diff
@@ -1,30 +1,33 @@
...
 });
 
-export const addNotification = (notification) => {
-  return {
-    type: 'ADD_NOTIFICATION',
-    notification
-  };
-};
-
-export const addNotificationSuccess = (notification) => {
```
Comment (d3mola) [inline src/redux/actionCreator/notificationsActions.js]: @albert i think the name `addNotification` is better than `addUsersNotification`. Same with the success an failure. This is up for debate though.

File: src/redux/middleware/approvalsSaga.js [hunk 9]
```diff
@@ -32,7 +32,7 @@ export function* updateRequestStatusSaga(action) {
     );
     toast.success(response.data.message);
     yield put(updateRequestStatusSuccess(response.data.updatedRequest));
-    yield put(closeModal());
+    // yield put(closeModal());
```
Comment (d3mola) [inline src/redux/middleware/approvalsSaga.js]: @albert85 please remove this comment if we dont need it

File: src/redux/middleware/notificationsSaga.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import { put, takeLatest, call } from 'redux-saga/effects';
+import { FETCH_USERS_NOTIFICATION, ADD_USERS_NOTIFICATION } from '../constants/actionTypes';
+import { fetchUsersNotificationSuccess, fetchUsersNotificationFailure, addUsersNotificationSuccess, addUsersNotificationFailure } from '../actionCreator/notificationsActions';
```
Comment (d3mola) [inline src/redux/middleware/notificationsSaga.js]: @albert85 please can we have each destructured import on a new line. Airbnb style.

File: src/components/RouterModal/index.jsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import RequestsModal from '../RequestsModal/RequestsModal';
+import Modal from '../modal/Modal';
+import { openModal, closeModal } from '../../redux/actionCreator/modalActions';
+
+class RouterModal extends Component {
+    state = {
+      viewRequestDetails: ''
+
```
[the remark this answers is not in the document]
Comment (albert85) [reply src/components/RouterModal/index.jsx]: done

File: src/components/Table/index.jsx [hunk 12]
```diff
@@ -130,7 +130,7 @@ export class Table extends Component {
           Status
         </th>
       </tr>
-    );
+    );
```
[the remark this answers is not in the document]
Comment (albert85) [reply src/components/Table/index.jsx]: done

File: src/components/modal/Modal.jsx (hunk 1 above)
Comment (albert85) [reply src/components/modal/Modal.jsx]: done

File: src/components/notification-pane/NotificationContainer.jsx (hunk 2 above)
Comment (albert85) [reply src/components/notification-pane/NotificationContainer.jsx]: done

File: src/components/notification-pane/NotificationItem.jsx (hunk 3 above)
Comment (albert85) [reply src/components/notification-pane/NotificationItem.jsx]: done

File: src/components/notification-pane/NotificationPane.jsx (hunk 6 above)
Comment (albert85) [reply src/components/notification-pane/NotificationPane.jsx]: done

File: src/redux/middleware/approvalsSaga.js (hunk 9 above)
Comment (albert85) [reply src/redux/middleware/approvalsSaga.js]: done

File: src/redux/middleware/notificationsSaga.js (hunk 10 above)
Comment (albert85) [reply src/redux/middleware/notificationsSaga.js]: done

**Comment by albert85** [comment]
@babadee001 PR is ready for your review

File: src/helper/socket/socket.js [hunk 13]
```diff
@@ -1,14 +1,15 @@
 import socketIOClient from 'socket.io-client';
-import toast from 'toastr';
-import store from '../../redux/store/store';
 import { addNotification } from '../../redux/actionCreator/notificationsActions';
+import store from '../../redux/store/store';
 
 const io = socketIOClient('http://127.0.0.1:5000');
```
Comment (amarachukwu-agbo) [inline src/helper/socket/socket.js]: Will this not affect the feature when it is moved to staging since the address would change? Should we not be getting the address from a .env file?
